cmplibrary/src/main/java/com/sourcepoint/cmplibrary/campaign/CampaignManager.kt

```diff
@@ -880,45 +881,85 @@ private class CampaignManagerImpl(
+                rejectedCategories = ccpaConsentStatus?.rejectedCategories
+            ),
             pubData = messageReq.pubData,
-            metaDataResp = metaDataResp,
-            ccpaMessageMetaData = ccpaMessageMetaData,
-            ccpaCS = ccpaConsentStatus,
-            usNatCS = null,
-            gdprMessageMetaData = null,
-            gdprCs = null,
+            messageId = ccpaMessageMetaData?.messageId,
+            sampleRateval = metaDataResp?.ccpa?.sampleRate
         )
     }
 
-    override fun getUsNatPvDataBody(messageReq: MessagesParamReq): JsonObject {
-        return toPvDataBody(
-            accountId = messageReq.accountId,
-            propertyId = messageReq.propertyId,
+    override fun getUsNatPvDataBody(messageReq: MessagesParamReq): PvDataRequest.USNat {
+        return PvDataRequest.USNat(
+            applies = metaDataResp?.usnat?.applies ?: false,
+            uuid = usNatConsentData?.uuid,
+            accountId = messageReq.accountId.toInt(),
+            siteId = messageReq.propertyId.toInt(),
+            consentStatus = com.sourcepoint.mobile_core.models.consents.ConsentStatus(
+                rejectedAny = usNatConsentData?.consentStatus?.rejectedAny,
+                consentedToAll = usNatConsentData?.consentStatus?.consentedToAll,
+                consentedToAny = usNatConsentData?.consentStatus?.consentedToAny,
+                granularStatus = com.sourcepoint.mobile_core.models.consents.ConsentStatus.ConsentStatusGranularStatus(
+                    sellStatus = usNatConsentData?.consentStatus?.granularStatus?.sellStatus,
+                    shareStatus = usNatConsentData?.consentStatus?.granularStatus?.shareStatus,
+                    sensitiveDataStatus = usNatConsentData?.consentStatus?.granularStatus?.sensitiveDataStatus,
+                    gpcStatus = usNatConsentData?.consentStatus?.granularStatus?.gpcStatus,
+                    defaultConsent = usNatConsentData?.consentStatus?.granularStatus?.defaultConsent,
+                    previousOptInAll = usNatConsentData?.consentStatus?.granularStatus?.previousOptInAll,
+                    purposeConsent = usNatConsentData?.consentStatus?.granularStatus?.purposeConsent
+                ),
+                hasConsentData = usNatConsentData?.consentStatus?.hasConsentData,
+                vendorListAdditions = usNatConsentData?.consentStatus?.vendorListAdditions,
+            ),
             pubData = messageReq.pubData,
-            metaDataResp = metaDataResp,
-            usNatCS = usNatConsentData,
-            ccpaMessageMetaData = null,
-            gdprMessageMetaData = null,
-            ccpaCS = null,
-            gdprCs = null,
+            sampleRate = metaDataResp?.usnat?.sampleRate,
+            msgId = usNatConsentData?.messageMetaData?.messageId,
+            categoryId = null,
+            subCategoryId = null,
+            prtnUUIDval = null
         )
```

Comment:
Why are `categoryId`, `subCategoryId` and `prtnUUID(val)` null?

Reply:
@andresilveirah previously, Android SDK sent a request without these values. In contrary, iOS sent a request with these values.

Reply:
I always check both implementation, when they disagree with each other, I check Postman and ultimately go with the iOS implementation, which I believe to be most correct.

Reply:
I'm fairly certain those values shouldn't be null.